lib/trino-filesystem/src/main/java/io/trino/filesystem/Location.java

```diff
@@ -61,9 +61,6 @@ public static Location of(String location)
         checkArgument(!location.isEmpty(), "location is empty");
         checkArgument(!location.isBlank(), "location is blank");
 
-        checkArgument(location.indexOf('#') < 0, "Fragment is not allowed in a file system location: %s", location);
-        checkArgument(location.indexOf('?') < 0, "URI query component is not allowed in a file system location: %s", location);
```

Comment:
is there anything else preventing us from using such paths correctly?
like some urlencode somewhere in delta?

would it make sense to have end-to-end tests with Delta, Hive, Iceberg?